pkg/resources/managed_account.go

```diff
@@ -0,0 +1,193 @@
+		ForceNew:     true,
+	},
+	"comment": &schema.Schema{
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies a comment for the managed account.",
+		ForceNew:    true,
+	},
+	"cloud": &schema.Schema{
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Cloud in which the managed account is located.",
+	},
+	"region": &schema.Schema{
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Snowflake Region in which the managed account is located.",
+	},
+	"locator": &schema.Schema{
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Display name of the managed account.",
+	},
+	"created_on": &schema.Schema{
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Date and time when the managed account was created.",
+	},
+	"url": &schema.Schema{
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "URL for accessing the managed account, particularly through the web interface.",
+	},
+}
+
+// ManagedAccount returns a pointer to the resource representing a managed account
+func ManagedAccount() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateManagedAccount,
+		Read:   ReadManagedAccount,
+		Delete: DeleteManagedAccount,
+		Exists: ManagedAccountExists,
+
+		Schema: managedAccountSchema,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+// CreateManagedAccount implements schema.CreateFunc
+func CreateManagedAccount(data *schema.ResourceData, meta interface{}) error {
+	return CreateResource(
+		"this does not seem to be used",
+		managedAccountProperties,
+		managedAccountSchema,
+		snowflake.ManagedAccount,
+		ReadManagedAccount,
+	)(data, meta)
+}
+
+// ReadManagedAccount implements schema.ReadFunc
+func ReadManagedAccount(data *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	id := data.Id()
+
+	stmt := snowflake.ManagedAccount(id).Show()
+	row := db.QueryRow(stmt)
+	var name, cloud, region, locator, createdOn, url, comment sql.NullString
+	var isReader bool
+	err := row.Scan(&name, &cloud, &region, &locator, &createdOn, &url, &isReader, &comment)
+	if err != nil {
+		return err
+	}
+
+	// TODO turn this into a loop after we switch to scaning in a struct
+	err = data.Set("name", name.String)
+	if err != nil {
+		return err
+	}
+	err = data.Set("cloud", cloud.String)
+	if err != nil {
+		return err
+	}
+
+	err = data.Set("region", region.String)
+	if err != nil {
+		return err
+	}
+
+	err = data.Set("locator", locator.String)
+	if err != nil {
+		return err
+	}
+
+	err = data.Set("created_on", createdOn.String)
+	if err != nil {
+		return err
+	}
+
+	err = data.Set("url", url.String)
+	if err != nil {
+		return err
+	}
+
+	if isReader {
+		err = data.Set("type", "READER")
+		if err != nil {
+			return err
+		}
+	} else {
+		return fmt.Errorf("is_reader should have been true but was %t", isReader)
```

Comment:
Unless I am missing something, this looks like it would break if and when snowflake introduced non-reader accounts?

Reply:
Yes, when more managed account types come in we would have to add them to the "type" field and add some logic here, something like:
```go
if isReader {
  // reader stuff
} else if isNewTypeOfAccount {
  // newType stuff
} else {
  // fail
}
```